aether-kernel/aether/kernel/api/models.py

```diff
@@ -118,37 +116,87 @@ def __str__(self):
 
     class Meta:
         app_label = 'kernel'
-        default_related_name = 'mappings'
+        default_related_name = 'mappingsets'
         ordering = ['project__id', '-modified']
         indexes = [
             models.Index(fields=['project', '-modified']),
             models.Index(fields=['-modified']),
         ]
 
 
+class Mapping(ExportModelOperationsMixin('kernel_mapping'), TimeStampedModel):
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4)
+    revision = models.TextField(default='1')
+    name = models.CharField(max_length=50, null=False, unique=True)
+    is_active = models.BooleanField(default=False)
+    definition = JSONField(blank=False, null=False)
+
+    @property
+    def definition_prettified(self):
+        return json_prettified(self.definition)
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        app_label = 'kernel'
+        default_related_name = 'mappings'
+        ordering = ['-modified']
+        indexes = [
+            models.Index(fields=['-modified']),
+        ]
+
+
+class MappingSetMapping(ExportModelOperationsMixin('kernel_mappingsetmapping'), TimeStampedModel):
```

Comment:
Couldn't we just have a many to many field in the `MappingSet`model? 

```python
    mappings = models.ManyToManyField(to=Mapping, blank=True)
```

Reply:
Not sure how django's `many-to-many` works, I just know a properly normalised db should not have a `many-to-many`. I will look into it

Reply:
I thin that the point is that Mapping can be reused in another MappingSet, so the question is, do we want that? or better, are we going to use that feature?

Reply:
Correct @obdulia-losantos it was an acceptance criteria.. 😄 About using it.... 🤷‍♂️